cnd/src/http_api/routes.rs

```diff
@@ -817,18 +912,12 @@ async fn handle_action_fund(id: LocalSwapId, facade: Facade) -> anyhow::Result<A
 
     let maybe_response = match role {
         Role::Alice => {
-            let state = facade.get_alice_herc20_halight_swap(id).await?;
-
-            state
-                .map(|state| state.fund_action().map(ActionResponseBody::from))
-                .flatten()
+            let swap = facade.get_alice_herc20_halight_swap(id).await?;
+            swap.fund_action().map(ActionResponseBody::from)
```

Comment:
Future, potential optimization:

It might be worthwhile to change the return value of `fund_action` to a `Result` so we can `?` here and do:

```suggestion
            let action = facade.get_alice_herc20_halight_swap(id).await?.fund_action()?;
            
            ActionResponseBody::from(action)
```

We would need to return a dedicated error from this that can be mapped to a `404`.

Reply:
That would also get rid of this line in every controller: `let response = maybe_response.ok_or(LndActionError::NotFound)?;`